docs/decisions/2022-07-test-fixtures.md

```diff
@@ -0,0 +1,90 @@
+# Prefer ephemeral test fixtures where possible
+
+* **Status:** approved
+* **Last Updated:** 2022-07
+* **Objective:** Clarify the use of permanent vs. reusable Cloud resources
+
+## Context & Problem Statement
+
+Emblem evolved to use long-lived, singleton test fixtures in end-to-end (E2E) test runs. These include resources such as Google Cloud projects, Cloud Storage buckets, container images in Artifact Registry, and Cloud Run services.
+
+We decided to investigate whether we should migrate these long-lived, singleton fixtures to a more ephemeral approach.
+
+## Priorities
+
+Ideally, our test fixtures would be designed such that:
+
+* Test infrastructure is quick to set up.
+* Test infrastructure is documented (e.g. via Terraform files).
+* Tests themselves are quick to write.
+* Both tests and their infrastructure require minimal maintenance.
+* Tests are runnable by multiple [teams of] developers in parallel.
+* Tests cover as much of the application surface as possible.
+* Tests allow developers and contributors to quickly identify root causes of bugs.
+* Tests have minimal reliance on existing state.
+
+Of course, some of these goals are mutually exclusive: supporting parallel test runs often increases the cost of writing (and in some cases maintaining) tests and test infrastructure.
+
+## Constraints
+
+### Some resources MUST be singletons
+While most of Emblem's Google Cloud resources are provisioned programmatically using Terraform, some (such as the Google Cloud projects themselves) are currently intended to be provisioned manually.
```

Comment:
non-blocking: This could be improved by explaining "why". I believe the answer is a blend:

* Because these things were prototyped in this way and the engineering investment to generalize from the prototype wasn't made
* Because organization policy constraints make spinning up ad hoc projects challenging.

Reply:
I would argue it's more the latter - the former is pretty easy to do IMO.

(In fact, I had done it that way in an earlier prototype! 🙂 )